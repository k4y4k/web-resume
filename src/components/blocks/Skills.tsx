import * as React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import Composed from '../skills/Composed'
import SectionContainer from '../section/SectionContainer'

interface SkillsItem {
  keywords: string[]
  name: string
}

interface SkillsList {
  skills: SkillsItem[]
}

export const PureSkills = ({ skills }: SkillsList): JSX.Element => {
  return (
    <div data-testid='skills' css='grid-area:skills;'>
      <SectionContainer title='Skills'>
        {skills?.map(el => (
          <Composed key={el.name} heading={el.name} bucket={el.keywords} />
        ))}
      </SectionContainer>
    </div>
  )
}

export const Skills = (): JSX.Element => {
  const data = useStaticQuery(graphql`
    {
      file(name: { eq: "data" }, extension: { eq: "json" }) {
        childDataJson {
          skills {
            name
            keywords
          }
        }
      }
    }
  `)

  const { skills } = data?.file.childDataJson

  return <PureSkills skills={skills} />
}

export default Skills
