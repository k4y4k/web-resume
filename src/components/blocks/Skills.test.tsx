import * as React from 'react'
import { PureSkills, Skills } from './Skills'
import { render, screen } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'

describe('<Skills />', () => {
  beforeAll(() =>
    (useStaticQuery as jest.Mock).mockReturnValue({
      file: {
        childDataJson: {
          skills: [
            {
              name: 'Web Development',
              keywords: ['HTML', 'CSS', 'Javascript'],
            },
          ],
        },
      },
    })
  )

  test('bails out on no data', () => {
    render(<PureSkills skills={[]} />)

    // still want the title
    expect(screen.getByTestId('skills')).toBeInTheDocument()

    expect(screen.queryByTestId('sectionItemContainer')).not.toBeInTheDocument()
  })

  test('renders array of >=1 skills items', () => {
    const data = [
      {
        name: 'Web Development',
        level: 'Master',
        keywords: ['HTML', 'CSS', 'Javascript'],
      },
    ]

    render(<PureSkills skills={data} />)

    const skillsList = screen.getAllByTestId('skills')

    expect(skillsList.length).toBe(1)
  })

  test('Skills renders OK', () => {
    render(<Skills />)

    expect(screen.queryByTestId('skills')).toMatchSnapshot()
  })
})
