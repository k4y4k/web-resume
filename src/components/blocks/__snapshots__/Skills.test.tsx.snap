// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> Skills renders OK 1`] = `
.c3 {
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.25rem;
}

.c2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.c1 {
  font-family: Inconsolata,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;
  font-size: 1.25rem;
  line-height: 1.75rem;
  font-weight: 700;
  text-transform: uppercase;
  --tw-text-opacity: 1;
  color: rgba(77,54,140,var(--tw-text-opacity));
}

.c0 {
  padding: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

<div
  data-testid="skills"
>
  <div
    class="c0"
    data-testid="sectionContainer"
  >
    <h2
      class="c1"
      data-testid="sectionTitle"
    >
      Skills
    </h2>
    <div
      class="c2"
      data-testid="skillsComposed"
    >
      <h1
        class="c3"
        data-testid="itemTitle"
      >
        Web Development
      </h1>
      <div
        data-testid="skillsBucket"
      >
        <span>
          HTML 
        </span>
        <span>
          CSS 
        </span>
        <span>
          Javascript 
        </span>
      </div>
    </div>
  </div>
</div>
`;
